plugin/audit_log_v2/audit_record.h

```diff
@@ -0,0 +1,59 @@
+/* Copyright (c) 2022 Percona LLC and/or its affiliates. All rights reserved.
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; version 2 of the License.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program; if not, write to the Free Software
+   Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA */
+
+#ifndef AUDIT_RECORD_V2_H_INCLUDED
+#define AUDIT_RECORD_V2_H_INCLUDED
+
+#include "mysql/plugin_audit.h"
+
+#include <memory>
+#include <string>
+
+namespace audit_log_v2 {
+
+/*
+ * Base audit record, contains elements mandatory for all records.
+ */
+struct AuditRecord {
+  std::string name;  // A string representing the type of instruction that
```

Comment:
Instead of using std::string directly define a typedef/usig for it. Initially it m ay be identical to std::string.
Later, when you decide to implement more efficient memory management (or resource monitoring), you will be able to change it to, say, std::pmr::string